src/Utilities/CommandLineBuilder.cs

```diff
@@ -242,16 +256,23 @@ virtual protected bool IsQuotingRequired(string parameter)
         }
 
         /// <summary>
-        /// Add a space to the specified string if and only if it's not empty. 
+        /// Add a space to the specified string if and only if it's not empty.
         /// </summary>
         /// <remarks>
         /// This is a pretty obscure method and so it's only available to inherited classes.
         /// </remarks>
         protected void AppendSpaceIfNotEmpty()
         {
-            if (CommandLine.Length != 0 && CommandLine[CommandLine.Length - 1] != ' ')
+            if (CommandLine.Length != 0)
```

Comment:
I _almost_ did that but you've convinced me to.  Its a protected method that I doubt anyone is overriding but hey.